pkg/broker/filter/filter_handler.go

```diff
@@ -235,10 +235,19 @@ func (h *Handler) handleDispatchToReplyRequest(ctx context.Context, trigger *eve
 	target := broker.Status.Address
 
 	reportArgs := &ReportArgs{
-		ns:          trigger.Namespace,
-		trigger:     trigger.Name,
-		broker:      trigger.Spec.Broker,
-		requestType: "reply_forward",
+		ns:            trigger.Namespace,
+		trigger:       trigger.Name,
+		broker:        trigger.Spec.Broker,
+		requestType:   "reply_forward",
+		requestScheme: request.URL.Scheme,
+	}
+
+	if reportArgs.requestScheme == "" {
+		if request.TLS != nil {
+			reportArgs.requestScheme = "https"
+		} else {
+			reportArgs.requestScheme = "http"
+		}
```

Comment:
nit: this is simpler in general, as `request.URL.Scheme` will always be empty
```suggestion
	}

	if request.TLS != nil {
		reportArgs.requestScheme = "https"
	} else {
		reportArgs.requestScheme = "http"
	}
```